sdk/core/azure-core/src/exception.cpp

```diff
@@ -28,4 +30,48 @@ namespace Azure { namespace Core {
     RawResponse = std::move(rawResponse);
   }
 
+  RequestFailedException::RequestFailedException(
+      std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse)
+      : std::runtime_error(getRawResponseField(rawResponse, "message"))
+  {
+    auto& headers = rawResponse->GetHeaders();
+
+    StatusCode = rawResponse->GetStatusCode();
+    ErrorCode = getRawResponseField(rawResponse, "code");
+    StatusCode = rawResponse->GetStatusCode();
+    ReasonPhrase = rawResponse->GetReasonPhrase();
+    RequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsRequestId);
+    ClientRequestId = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::MsClientRequestId);
+    Message = this->what();
+  }
+
+  std::string RequestFailedException::getRawResponseField(
+      std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse,
+      std::string fieldName)
+  {
+    auto& headers = rawResponse->GetHeaders();
+    std::string contentType = HttpShared::GetHeaderOrEmptyString(headers, HttpShared::ContentType);
+    std::vector<uint8_t> bodyBuffer = rawResponse->GetBody();
+    std::string result;
+
+    if (contentType.find("json") != std::string::npos)
+    {
+      auto jsonParser = Azure::Core::Json::_internal::json::parse(bodyBuffer);
+      auto error = jsonParser.find("error");
+      if (error != jsonParser.end() && error.value().contains(fieldName))
+      {
+        result = error.value()[fieldName].get<std::string>();
```

Comment:
@vhvb1989 will the error always be a string, or could it be a number?

Reply:
string from json. At least for Key Vault. Each service will own its error response payload

Reply:
According to @JeffreyRichter, Azure defines errors as follows, so string should be the right general purpose structure/JSON schema:
```json
{
  "error": {
    "code": "ServiceDefinedErrorCode",
    "message": "Human-readable description",
    "target": "target of error",
    "innererror": {
      "code": "PasswordTooShort",
      "minLength": 6,
    }
  }
}
```